app/javascript/articles/components/__tests__/SaveButton.test.jsx

```diff
@@ -0,0 +1,87 @@
+import { h } from 'preact';
+import { fireEvent, render } from '@testing-library/preact';
+import { axe } from 'jest-axe';
+import { SaveButton } from '../SaveButton';
+
+it('should have no a11y violations', async () => {
+  const article = { class_name: 'Article', id: 1 };
+  const { container } = render(<SaveButton article={article} />);
+  const results = await axe(container);
+
+  expect(results).toHaveNoViolations();
+});
+
+it('should render button as bookmarked', () => {
```

Comment:
@dengaku12, the next two tests are just checking to see if the correct button text is loading based on whether or not an article is bookmarked.